src/main/kotlin/org/jitsi/nlj/dtls/DtlsStack.kt

```diff
@@ -86,12 +88,7 @@ class DtlsStack(
      */
     var eventHandler: EventHandler? = null
 
-    private var running: Boolean = false
-    private val incomingProtocolData = java.util.ArrayDeque<ByteBuffer>(QUEUE_SIZE)
-    /**
-     * This lock is used to make access to [running] and [incomingProtocolData] atomic
-     */
-    private val lock = Any()
+    private val incomingProtocolData = ArrayBlockingQueueWithShutdown<ByteBuffer>(50)
```

Comment:
Do we not still have `QUEUE_SIZE`?  Also I think we may want to pass `true` in the constructor to make it fair.

Reply:
I put `QUEUE_SIZE` back.

Do we ever have more than one queue listener at a time, such that fairness matters?  The old `ArrayBlockingQueue`-based implementation didn't have fairness.

Reply:
I'm not sure if we end up with multiple listeners, but it seems like a sensible thing to do for the use case.  I see that we weren't setting this param before, but I feel like we probably should have been just in case.